### PR TITLE
[Tests 1/4] chore(tests): update project structure to support upstream test suite

### PR DESCRIPTION
**Pull request by ephread**
## Checklist

- [x] The new code additions passed the tests (`npm test`).
- [x] The linter ran and found no issues (`npm run-script lint`).

## Description

Following #489.

Sorry about the noise, the editor removed trailing spaces. Changes are:

- `common.js` was updated to support the `category` parameter;
- specs were moved into the `inkjs` subdirectory;
- `tests.ink` and `tests.json` were respectively moved to `files/original/inkjs` and `files/compiled/inkjs`; note that `tests.json` was renamed to `tests.ink.json`, it's pure convenience, since the compilation script (I'll add it in another PR) does not strip `.ink` from the output.

Let me know what you think, even on petty changes like `Strip the BOM.` 😁

### CODE REVIEW

File: tests/common.js [hunk 1]
```diff
@@ -1,24 +1,33 @@
-var inkPath = '../engine/runtime.js'
+let path = require('path');
+
+var rootDir = path.join(__dirname, '..');
+var inkPath = path.join(rootDir, 'engine', 'runtime.js');
```
Comment (y-lohse) [inline tests/common.js:4]: Nit, but we'd better use `let` everywhere while we're at it.

Comment (ephread) [reply tests/common.js:4]: We can use `let` everywhere. I hardly ever write `var` anyway, I don't know why it's there to be honest 😅. I'll update the PR!

**Comment by ephread** [comment]
All fixed!

**Comment by ephread** [comment]
CI seems to have issues fetching the repository at the moment. I'll merge the PR once everything is green.

**Pull request merged**